### PR TITLE
Avoid Timeout and Threads when not needed

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: lib/rubysl/timeout.rb [hunk 1]
```diff
@@ -120,13 +120,22 @@ def self.watch_channel
...
+  CONTROLLER_MUTEX = Mutex.new
+
+  def self.ensure_timeout_thread_running
+    unless @controller
+      CONTROLLER_MUTEX.synchronize do
+        unless @controller
+          @controller = Thread.new { watch_channel }
+        end
+      end
+    end
```
Comment (chrisseaton) [inline lib/rubysl/timeout.rb]: Is this double-checked locking? That's only safe in Java if your variable is volatile, isn't it? How does it translate to Ruby?

Comment (eregon) [reply lib/rubysl/timeout.rb]: Shared writes are volatile with thread-safe objects.
But anyway, this `@controller` variable is never read here, it only keeps the Thread instance alive and is used as a spawned/not-spawned marker.

Comment (chrisseaton) [reply lib/rubysl/timeout.rb]: This seems like a premature optimisation. I'd normally write this as a simple lock, but if you think it's ok then I'm fine to commit.

Comment (eregon) [reply lib/rubysl/timeout.rb]: You are right, it is premature here, I did not measure and there is anyway cross-thread interactions around, I'll remove it.

**Pull request merged**